### PR TITLE
Add client request pathway

### PR DESCRIPTION
**Pull request by SHeimlich**
Create pathway for Arcs DevTools client to send messages to the Arcs DevToolsService and have the service respond. PR includes:

- Creation of onMessageCallbacks in DevToolsService
- Add an onMessageCallback to DevToolsService to handle storagekey requests
- Add a button to client to request storage

cc: @piotrswigon

### CODE REVIEW

File: java/arcs/android/devtools/DevToolsService.kt [hunk 1]
```diff
@@ -98,6 +99,17 @@ open class DevToolsService : Service() {
         devToolsServer.send(service.storageKeys ?: "")
       }
 
+      devToolsServer.addOnMessageCallback { message ->
```
Comment (piotrswigon) [inline java/arcs/android/devtools/DevToolsService.kt:102]: Should you ever remove this callback? What will happen if user toggles DevTools off? If we don't yet handle it gracefully we should probably file a bug. And we should probably have an umbrella bug will all the potential DevTools related fixes we should someday do, I don't think we have set it up yet.

You can also ping me directly and we can set that up together.

**Review by piotrswigon — APPROVED** [review]

**Review by jasonwyatt — APPROVED** [review]

File: java/arcs/android/devtools/DevToolsService.kt (hunk 1 above)
Comment (jasonwyatt) [reply java/arcs/android/devtools/DevToolsService.kt:102]: +1

Comment (SHeimlich) [reply java/arcs/android/devtools/DevToolsService.kt:102]: Parent bug is b/154554631 added a sub-bug to track this: b/170155112

File: java/arcs/android/devtools/DevToolsService.kt [hunk 2]
```diff
@@ -98,6 +99,17 @@ open class DevToolsService : Service() {
...
+        val json = Json.parse(message)
+        when (json) {
+          is JsonValue.JsonObject -> {
+            if (json["type"].value == "request" && json["message"].value == "storageKeys") {
+              devToolsServer.send(service.storageKeys ?: "")
```
[the remark this answers is not in the document]
Comment (SHeimlich) [reply java/arcs/android/devtools/DevToolsService.kt]: Added socket to onMessage callbacks and a server.send method that lets you specify a socket. Opened b/170154700 to track testing.

**Pull request closed without merging**